website/docs/using-odo/create-component.md

```diff
@@ -3,4 +3,139 @@ title: Create Component
+    ```
+   By default, `java-maven` is selected for us. Since this is a Spring Boot application, we should be selecting `java-springboot`. 
+
+    We can either scroll down to `java-springboot` using the arrow key, or start typing `spring` on the prompt. Typing `spring` will lead to odo filtering the component type based on your input.
+
+2. Next, odo asks you to name the component:
+    ```shell
+    odo create                
+    ? Which devfile component type do you wish to create java-springboot
+    ? What do you wish to name the new devfile component (java-springboot) petclinic
+    ```
+    Let's name it `petclinic`.
+
+3. Next, odo asks you for the project in which you would like to create the component. Use the project `myproject` that we created earlier or the one you had set using `odo project set` command
+   ```shell
+   odo create
+   ? Which devfile component type do you wish to create java-springboot
+   ? What do you wish to name the new devfile component java-springboot
+   ? What project do you want the devfile component to be created in myproject
+   ```
+   Now you will have a `devfile.yaml` in your current working directory. But odo is just not done asking you questions yet.
+4. Lastly, odo asks you if you would like to download a starter project. Since we already cloned the petclinic project, we answer in No by typing `n` and hitting the return key:
+   ```shell
+   odo create
+   ? Which devfile component type do you wish to create java-springboot
+   ? What do you wish to name the new devfile component java-springboot
+   ? What project do you want the devfile component to be created in myproject
+   Devfile Object Validation
+   ✓  Checking devfile existence [66186ns]
+   ✓  Creating a devfile component from registry: stage [92202ns]
+   Validation
+   ✓  Validating if devfile name is correct [99609ns]
+   ? Do you want to download a starter project (y/N) n
+   ```
+   
+Your Spring Boot component is now ready for use.
+
+## Non-interactively creating a component
+
+To non-interactively create the same component as we created in interactive mode, `cd` into the cloned petclinic project and execute:
+```shell
+odo create java-springboot petclinic -n myproject
+```
+Here `java-springboot` is the type of the component, `petclinic` is the name of the component, and `-n myproject` tells odo to use the project `myproject` for the mentioned `odo create` operation.
+
+## Starter projects
+
+Besides creating a component for an existing code, you could also use "starter project" when creating a component.
+
+Starter projects are example projects developed by the community to showcase the usability of devfiles. An odo user can use these starter projects by running `odo create` command in an empty directory.
+
+### Starer projects in interactive mode
+
+To interactively create a Java Spring Boot component using the starter project, you can follow the below steps:
+```shell
+mkdir myOdoComponent && cd myOdoComponent
+$ odo create
```

Comment:
Would `&&` work on windows/mac? If not, should we separate the 2 commands?
Also, it would help with copying the command if you separate `odo create`  in a different code block.